skins/laika/tests/unit/components/pages/donation_form/PaymentType.spec.ts

```diff
@@ -0,0 +1,42 @@
+import { createLocalVue, mount } from '@vue/test-utils';
+import Vuex from 'vuex';
+import Buefy from 'buefy';
+import PaymentType from '@/components/pages/donation_form/PaymentType.vue';
+import { createStore } from '@/store/donation_store';
+import { action } from '@/store/util';
+import { NS_PAYMENT } from '@/store/namespaces';
+import { setType } from '@/store/payment/actionTypes';
+
+const localVue = createLocalVue();
+localVue.use( Vuex );
+localVue.use( Buefy );
+
+const testPaymentMethods = [
+	{ id: 'debit-card', type: 'BEZ' },
+	{ id: 'paypal', type: 'PPL' },
+	{ id: 'bank-transfer', type: 'UEB' },
+];
+
+describe( 'PaymentType', () => {
+
+	it( 'sends new interval to store when amount is selected', () => {
+		const wrapper = mount( PaymentType, {
+			localVue,
+			propsData: {
+				paymentTypes: testPaymentMethods,
+			},
+			store: createStore(),
+			mocks: {
+				$t: () => {},
+			},
+		} );
+		const store = wrapper.vm.$store;
+		store.dispatch = jest.fn();
+
+		wrapper.find( '#debit-card' ).trigger( 'click' );
+		const expectedAction = action( NS_PAYMENT, setType );
+
+		expect( store.dispatch ).toBeCalledWith( expectedAction, 'BEZ' );
```

Comment:
I think it might be better if we do `wrapper.find( '#' + testPaymentMethods[0].id)` and then `expect( store.dispatch ).toBeCalledWith( expectedAction, testPaymentMethods[0].type) );`

Reply:
I have spent quite some time going back and forth between those versions and a third version, where I extract the data to view_model and use it in the test. There are tradeoffs for both versions:

* The "plain strings and values" version conforms to the principle that tests should state plain and simple what the values are, without complicated assembly or data from external sources. It also demonstrates that the component works independently from the properties that are passed in (I could have passed in 2-month intervals etc). 
* The "composed expected values"/"external data" version has the benefit of using the "real" data that the user sees, with less risk that tests miss something. It can also make the test more expressive, by explicitly stating what, for example, `year` and `12` *mean* (`id` and `intervalInMonths`). 

If you like the "composed expected data" version better, I'll gladly change the tests.

Reply:
Oh ok, I don't mind, I just thought that it's harder to see that BEZ corresponds to debit card when they are plain strings, because a lot of the language in our domain is not common knowledge.